States in CT - i18n doc (#5733)
vuestorefront/vue-storefront: packages/core/docs/commercetools/configuration.md

}]
```

- `useNuxtI18nConfig` - when enabled, `@vue-storefront/commercetools/nuxt` package will use `i18n` config object provided in `nuxt.config.js`. Otherwise, the `i18n` config should be passed directly to this package configuration. You can read more about it on [Internationalization](../advanced/internationalization.md) page. There are things specific to commercetools, which you can read more about below.

### Adding states for the country

> In commercetools, each state must have a specified tax rule. In order to add a new state, add `states` field in the desired country inside `i18n.countries` in `nuxt.config.js`:

Reviewer: ```suggestion
In order to add a new state, add `states` field in the desired country inside `i18n.countries` in `nuxt.config.js`:
```

```js
i18n: {
  countries: [
    { name: 'US',
      label: 'United States',
      states: [
        'California',
        'Nevada'
      ]
    },
    // ...
  ]
}
```

> Then in commercetools' dashboard, open *Settings* -> *Project settings* -> *Taxes* and click **Add rate** button for new states:

Reviewer: I will put whole part with tax rules as a toggle and just add warning that each state needs a tax rule. 

You describe it as it was a natural part of adding a new state to the config and you assumed that the tax rule is not assigned. I think there is a context mismatch VSF/CT configuration. It's good to mention that it's needed but it shouldn't interrupt the reading flow because you don't know how it was configured previously

Reviewer 2: hmm, I suggest putting just the links to the Commercetools about the configuring states - that's it.

Author: Ok, I've put it in the toggle

}]
```

> - `useNuxtI18nConfig` - when this property is set to true, `@vue-storefront/commercetools/nuxt` package will use `i18n` config object provided in `nuxt.config.js`. When set to false, `i18n` config should be declared directly inside this package configuration. You can read more generally about `i18n` config in Vue Storefront [here](../advanced/internationalization.md). On the other hand, there are things which are specific for the commercetools, you can read about them below.

Reviewer 3: ```suggestion
- `useNuxtI18nConfig` - when enabled, `@vue-storefront/commercetools/nuxt` package will use `i18n` config object provided in `nuxt.config.js`. Otherwise, the `i18n` config should be passed directly to this package configuration. You can read more about it on [Internationalization](../advanced/internationalization.md) page. There are things specific to commercetools, which you can read more about below.
```

> ### Adding states for the country

Reviewer: What I'm missing is

- the use case: "Adding states for the country" is not a use case because I don't know the context (checkout/profile etc)
-  there is no info where I can access this data in my app, there is only a mention about a single usage of this data on the UI ut no mention on how to retrieve this data actually (will it be available in some composales or through $i18n from useContext etc)

> In commercetools each state must have a specified tax rules. That's why we decided to have the state as a select field instead of a text one.
> 
> In order to add a support for the new state, you have to:
> - Add `states` field in the desired country inside a `i18n.countries` of your `nuxt.config.js`

Reviewer 3: ```suggestion
In commercetools, each state must have a specified tax rule. In order to add a new state, add `states` field in the desired country inside `i18n.countries` in `nuxt.config.js`:
```

```js
i18n: {

>   currency: 'USD',
>   country: 'US',

Reviewer: ```suggestion
```

  countries: [
    { name: 'US',
      label: 'United States',
      states: [
        'California',
        'Nevada'
      ]
    },
    // ...
  ]
}
```

> - In commercetools' dashboard, open *Settings* -> *Project settings* -> *Taxes* and click **Add rate** button for new states:

Reviewer 3: ```suggestion
Then in commercetools' dashboard, open *Settings* -> *Project settings* -> *Taxes* and click **Add rate** button for new states:
```

![settings taxes for states](./../images/ct-taxes.png)

> After that, if you select `United States` as a country, there will be 2 available states. You can check it on Checkout's shipping & billing step and MyAccount's user shipping & billing addresses views.

Reviewer 3: ```suggestion
If you select `United States`, there will be 2 available states. You can check it on Checkout's shipping and billing steps and MyAccount's user shipping and billing address views.
```

}]
```

- `useNuxtI18nConfig` - when enabled, `@vue-storefront/commercetools/nuxt` package will use `i18n` config object provided in `nuxt.config.js`. Otherwise, the `i18n` config should be passed directly to this package configuration. You can read more about it on [Internationalization](../advanced/internationalization.md) page. There are things specific to commercetools, which you can read more about below.

### Adding available states to the country in Checkout and MyAccount
In order to add a new state, add `states` field in the desired country inside `i18n.countries` in `nuxt.config.js`:
```js
i18n: {
  countries: [
    { name: 'US',
      label: 'United States',
      states: [
        'California',
        'Nevada'
      ]
    },
    // ...
  ]
}
```

> ::: warning Each state requires the tax rules
> A state without tax rules cannot work properly. Make sure to add the tax rules for each state in the commercetools.
> :::
> 
> :::details How to add tax rules to the state?
> Information about setting tax rules in the commercetools can be found [there](https://docs.commercetools.com/merchant-center/project-settings#configuring-taxes).
> :::

Reviewer 3: WDYT about merging these two notifications into one?

```suggestion
::: warning Each state requires the tax rules
A state without tax rules cannot work properly. Make sure to add the tax rules for each state in the commercetools as described [here](https://docs.commercetools.com/merchant-center/project-settings#configuring-taxes).
:::
```

Author: Good one

> After applying tax rules, if you select `United States`, there will be 2 available states. You can check it on Checkout's shipping and billing steps and MyAccount's user shipping and billing address views.

Reviewer 3: ```suggestion
If you select `United States`, after applying tax rules, there will be 2 states available. You can now select them on Checkout's shipping and billing steps and MyAccount's user shipping and billing address views.
```

}]
```

- `useNuxtI18nConfig` - when enabled, `@vue-storefront/commercetools/nuxt` package will use `i18n` config object provided in `nuxt.config.js`. Otherwise, the `i18n` config should be passed directly to this package configuration. You can read more about it on [Internationalization](../advanced/internationalization.md) page. There are things specific to commercetools, which you can read more about below.

### Adding available states to the country in Checkout and MyAccount
In order to add a new state, add `states` field in the desired country inside `i18n.countries` in `nuxt.config.js`:
```js
i18n: {
  countries: [
    { name: 'US',
      label: 'United States',
      states: [
        'California',
        'Nevada'
      ]
    },
    // ...
  ]
}
```
::: warning Each state requires the tax rules
A state without tax rules cannot work properly. Make sure to add the tax rules for each state in the commercetools.
:::

:::details How to add tax rules to the state?

> In commercetools' dashboard, open *Settings* -> *Project settings* -> *Taxes* and click **Add rate** button to add tax rates for new states:

Reviewer 2: just put the link to the docs, don't create tutorial in our docs for 3rd party service
